utils/paramfetch/src/lib.rs

```diff
@@ -48,18 +48,23 @@ struct ParameterData {
 }
 
 #[inline]
-fn param_dir() -> String {
-    std::env::var(DIR_ENV).unwrap_or_else(|_| PARAM_DIR.to_owned())
+fn param_dir(mut data_dir: PathBuf) -> String {
```

Comment:
Why is there so much cloning whenever `param_dir` is invoked? IIUC the lazy part will be executed at most once so better allocate once only there, the rest should get passed by reference. Or am I missing something?

Another thing, this function doesn't seem to be `inline`-friendly anymore.

Reply:
clippy didn't like having &PathBuf

Reply:
Done through &Path